src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -68,7 +68,15 @@ sealed abstract class ArraySeq[+A]
     ArraySeq.unsafeWrapArray(dest).asInstanceOf[ArraySeq[B]]
   }
 
-  override def map[B](f: A => B): ArraySeq[B] = iterableFactory.tabulate(length)(i => f(apply(i)))
+  override def map[B](f: A => B): ArraySeq[B] = {
+    val a = new Array[Any](size)
+    var i = 0
+    while (i < a.length){
+      a(i) = f(apply(i)).asInstanceOf[Any]
+      i += 1
+    }
+    ArraySeq.unsafeWrapArray(a).asInstanceOf[ArraySeq[B]]
```

Comment:
I'm confused, looking at the `updated` above and `prepended`/`appended` below why everything casts to Any and then casts back to the right type...  Is it because otherwise every statement, e.g. `a(i) = f(apply(i))`, gets casts inserted into it?  Just thinking out loud.

Reply:
The `asInstanceOf[Any]` in the `f(apply(i)).asInstanceOf[Any]` is unnecessary. I will remove it. 

All the methods instanciate an Array[Any], because they can't instantiate an Array[B]. They would need a `ClassTag[B]` to instantiate of `Array[B]`, but the methods as defined in `Seq` (and it's ancestors) don't provide it. This is also the reason why unlike `Array`/`ArrayOps`, `ArraySeq` looses primitive type specialisation when modified.

Reply:
Ah, right, thanks.